src/security.c

```
@@ -1,1012 +1,616 @@
+        memcpy(out_private_key, n_priv, PRIVATE_KEY_LENGTH);
+        return 0;
     }
 
-    /* Update secure element counter (high bits only) */
-    uint32_t high_bits = (uint32_t)(security_state.primary_counter.value >> 32);
-    if (high_bits > (uint32_t)(security_state.secure_counter.value >> 32))
-    {
-        security_state.secure_counter.value = security_state.primary_counter.value;
-        security_state.secure_counter.timestamp = security_state.primary_counter.timestamp;
-        security_state.secure_counter.valid = 1;
+    uint8_t priv[PRIVATE_KEY_LENGTH];
```

Clang (14.0.6)

[warning] variable 'priv' is not initialized

(cppcoreguidelines-init-variables)